app/src/main/java/com/scouts/kitchenplaner/model/usecases/ShowPersonalStartScreen.kt

```diff
@@ -35,22 +41,42 @@ class ShowPersonalStartScreen @Inject constructor(
 ) {
 
 
-    suspend fun getLatestProjectsForCurrentUser(): List<ProjectStub> {
-        val currentUser = dataStore.getCurrentUser()
-        return projectRepository.getLastShownProjectIds(user = currentUser, limit = AMOUNT_PROJECTS)
-            .map {
-                projectRepository.getProjectMetaDataByID(it).map { metadata ->
-                    metadata.stub
-                }.first()
+    fun getLatestProjectsForCurrentUser(): Flow<List<ProjectStub>> {
+        var currentUser: User
+        runBlocking {
+            currentUser = dataStore.getCurrentUser()
+        }
+        return projectRepository.getLastShownProjectIds(currentUser, AMOUNT_PROJECTS)
+            .flatMapLatest { ids ->
+                val helper = ids.map { id ->
+                    val project = projectRepository.getProjectMetaDataByID(id).map { it.stub }
+                    val projectFlow: Flow<List<ProjectStub>> = flowOf(listOf())
+                    projectFlow.combine(project) { prev, prod ->
+                        return@combine prev + prod
+                    }
```

Comment:
What exactly does the ```projectFlow``` flow do? To me it seems you are packing each project into a (one-element) list, only to later flatten a list consisting of these one-element list into a normal list. Shouldn't ```helper = ids.map { id -> repo.getMetaData(...) }``` and then in the combine (line 58) ```it.toList()``` achieve the same thing while being way more readable?